cep-recipe-serialization.md

```diff
@@ -0,0 +1,1117 @@
+# Recipe format: Rendered recipe YAML and serialization
+
+## Motivation
+
+We want to specify everything that goes into the final package in a standardized
+format. This defines the structure and contents of the rendered_recipe.yaml file
+which captures the complete recipe, build configuration, dependencies and
+sources used to create a conda package.
+
+The goal is to have a single file that contains all the information needed to
+reproduce an exact build of a package. This will enable better reproducibility,
+provenance tracking, and debugging of conda packages.
+
+## Storage
+
+The rendered recipe YAML file is stored in the `info/recipe` folder of the final
+package. Alongside the rendered recipe, we copy the entire "recipe folder".
+If the original recipe was not named `recipe.yaml`, then we rename it inside the package.
+
+For example:
+
+```txt
+├── recipe
+    ├── patches
+    │   └── patch1.diff
+    ├── build.sh
+    ├── build.bat
+    └── my_recipe.yaml
+```
+
+Will be copied into the following folder structure in the final package:
+
+```txt
+├── ...
+├── index.json
+└── recipe
+    ├── build.bat
+    ├── build.sh
+    ├── recipe.yaml
+    ├── patches
+    │   └── patch1.diff
+    ├── rendered_recipe.yaml
+    └── variant_config.yaml
+```
+
+## Specification
+
+The rendered recipe YAML file contains the following top-level sections:
```

Comment:
Should this be versioned too? Add a `rendered_recipe_version: 1` or something?

Reply:
I did not!